### ISSUE TITLE
auth0.WebAuth.renewAuth time out in Firefox and IE11

### ISSUE THREAD

**Issue by blong824** [auth0/auth0.js#564]
I am using auth0-js 8.8.0.  In Firefox and IE11 whenever I call the renewAuth function it waits 60 seconds and I get a timeout message.  The same behavior happens with 8.10.1.  However when I use 8.11.2 all browers fail after the timeout.

I am able to authorize and login.  This only happens on renewAuth.

In Firefox the iFrame is added to the page but the body is empty. 

I have an angular 2 app using 4.4.0.

My renew token client code looks like:
```
 public renewToken(): Observable<AuthResult> {
    const self = this;

    const auth0Config = {
      audience: 'pcc-profiles',
      redirectUri: this.tokenRefreshUri,
      usePostMessage: true};

    return Observable.create(observer => {
      self.auth0.renewAuth(auth0Config, (err, result) => {
        if (err) {
          console.log('rae: ', err);
          observer.error('renew auth token error: ', err);
        } else if (result && result.accessToken) {
          observer.next(result);
          observer.complete();
        }
      });
    });
```

The silent page from the server is pasted below:
```
@GetMapping(path="/", produces="text/html")
	public @ResponseBody String serveHiddenIframePage() {
		logger.debug("serveHiddenIframePage() to - " + tokenRenewalEndpoint);
		
		return "<!doctype html>"
		+ "<html>"
		+ "<head>"
		+  "<meta charset=\"utf-8\">"
		+  "<script src=\"https://cdn.auth0.com/js/auth0/8.8/auth0.min.js\"></script>"
		+  "<script>"
		+    "var webAuth = new auth0.WebAuth({"
		+      "domain: 'paycheckcity.auth0.com',"
		+      "clientID: 'b0h5ZpLfDSPRvCmKot0Rzgk2xJgx7Res',"
		+      "scope: 'openid profile',"
		+      "responseType: 'token id_token',"
		+      "redirectUri: '" + tokenRenewalEndpoint + "'"
		+    "});"
		+  "</script>"
		+  "<script>"
		+    "webAuth.parseHash(window.location.hash, function (err, response) {"
		+      "parent.postMessage(err || response, '" + tokenRenewalEndpoint + "');"
		+    "});"
		+  "</script>"
		+"</head>"
		+"<body></body>"
		+"</html>";
	}
```

**Comment by luisrudge** [comment]
This normally happens when the iframe page has an error. Can you debug your web traffic to see what is being rendered inside the iframe?

**Comment by blong824** [comment]
It doesn't look like anything it being rendered inside the iframe.  Both the head and body are empty tags.

**Comment by luisrudge** [comment]
Before rendering your callback page, the /authorize page from Auth0 is rendered in the iframe. We need to know if that page is throwing an error. Can you get the resulting authorize url (starts with https://yourdomain.auth0.com/authorize) and open that specific url in IE11?

**Comment by blong824** [comment]
i tried in it firefox, i dont have access to ie11 right now.  here is the resulting message: 
Failed to execute ‘postMessage’ on ‘DOMWindow’: The target origin provided (‘http://localhost:4200’) does not match the recipient window’s origin (‘http://localhost:8080’). 
I opened: https://paycheckcity.auth0.com/authorize?client_id=...

**Comment by luisrudge** [comment]
Your `tokenRenewalEndpoint` is probably wrong. What's your setup? Do you have two web servers running (8080 and 4200)?

**Comment by blong824** [comment]
Yes, I have the angular app running on 4200.  I have the server on 8080.  The tokenRenewalEndpoint is http://localhost:4200/callback.  This does work on Chrome.

**Comment by luisrudge** [comment]
Maybe chrome has a more relaxed validation? Not sure. Can you try to change the second parameter to be ‘http://localhost:8080’?

**Comment by blong824** [comment]
That is what it is set to.

**Comment by blong824** [comment]
This seems to be a similar issue: [issue](https://community.auth0.com/questions/2909/angular2-sso-using-webauthrenewauth-for-silent-cal)

**Comment by blong824** [comment]
After some more debugging, I see that the call to https://yourdomain.auth0.com/authorize is executed.  Right after that call, the redirect_uri from this call is then called and the content of the iframe from my server on port 8080 is fetched.  The src element of the hidden iframe is  https://yourdomain.auth0.com/authorize.  It just seems like it is not being executed.

**Comment by luisrudge** [comment]
I think it is being executed, but the postMessage origin validation fails. The error message says that the origin is wrong, so can you try to switch from 8080 to 4200? Only in the `parent.postMessage` call.

Also, there's an improved way of getting new tokens that doesn't require you to host a dedicated page for the callback. It requires an easy setup in the dashboard and is the best way to renew a token moving forward. If you want to test, there's more docs here: https://auth0.com/docs/libraries/auth0js/v8#using-checksession-to-acquire-new-tokens

**Comment by blong824** [comment]
Trying checkSession.  Keep getting a 500 error, Unable to construct sso user.  Probably not setting it up correctly?

**Comment by blong824** [comment]
Just wanted to update you on this.  Going back to using renewAuth....We have the same issues on our test and production environments.  These environments have the same client and server domains.  The other point of interest is when I update to auth0-js to 8.11.2 all browsers including chrome stop working.  Is there anything specific between versions that add more validation or anything that might help us to solve what is happening in Firefox?  Thanks

**Comment by blong824** [comment]
I took your advice and started using checkSession.  It is now working in Chrome and Firefox.  I wil be testing IE as well.  Thank you for your help and super fast responsiveness. :)
